astropy/utils/xml/writer.py

```diff
@@ -148,6 +153,20 @@ def start(self, tag, attrib={}, **extra):
         return len(self._tags)
 
     @contextlib.contextmanager
+    def xml_escaping(self, enabled=True, clean_kwargs=None):
+        current_xml_escape_cdata = self.xml_escape_cdata
+        if not enabled:
+            if HAS_BLEACH:
+                if clean_kwargs is None:
+                    clean_kwargs = {}
+                self.xml_escape_cdata = lambda x: bleach.clean(x, **clean_kwargs)
+            else:
+                raise ValueError('bleach package is required when HTML escaping is disabled.\n'
+                                 'Use "pip install bleach".')
```

Comment:
Good point that it should have a docstring.  

>so it doesn't matter if it doesn't work if its dependencies aren't met.

By "doesn't work", do you mean raises an exception?  That would be a problem in the current use case.

> It just seems strange to me to have a function that takes a flag that basic means "this should work" or "this shouldn't work".

I'm not sure what you mean by "shouldn't work".  The flag means:

-  `True`: Use the default fast brute-force sanitizing in `XMLWriter` by replacing a certain set of characters.  No XML will survive.
- `False`: Use nuanced (slower) sanitizing by actually parsing and cleaning the XML with bleach.  Whitelisted XML will survive.  We've agreed that if the user wants any whitelisted XML via this option that they must have bleach installed.